### PR TITLE
Upgrade dioxus to 0.5

### LINKED ISSUE #3: feat: Dioxus 0.5 support
Will update all the links and all that as well

### PR DESCRIPTION
**Pull request by logandeancall**
Closes https://github.com/dioxus-community/dioxus-helmet/issues/3

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,10 +1,10 @@
 //! ## General
-//! Inspired by react-helmet, this small [Dioxus](https://crates.io/crates/dioxus) component allows you to place elements in the **head** of your code.
+//! Inspired by react-dioxus-helmet, this small [Dioxus](https://crates.io/crates/dioxus) component allows you to place elements in the **head** of your code.
 //!
 //! ## Configuration
 //! Add the package as a dependency to your `Cargo.toml`.
 //! ```no_run
-//! cargo add dioxus-helmet
+//! cargo add dioxus-dioxus-helmet
```
Comment (marc2332) [inline src/lib.rs]:
```suggestion
//! cargo add dioxus-helmet
```

File: src/lib.rs [hunk 2]
```diff
@@ -16,32 +16,32 @@
...
+//!     }
 //! }
 //! ```
 //!
 //! Reach your dynamic values down as owned properties (eg `String` and **not** `&'a str`).
 //!
 //! Also make sure that there are **no states** in your component where you use Helmet.
 //!
-//! Any children passed to the helmet component will then be placed in the `<head></head>` of your document.
+//! Any children passed to the dioxus-helmet component will then be placed in the `<head></head>` of your document.
```
Comment (marc2332) [inline src/lib.rs]:
```suggestion
//! Any children passed to the `Helmet` component will then be placed in the `<head></head>` of your document.
```

File: src/lib.rs [hunk 3]
```diff
@@ -93,41 +88,6 @@ pub fn Helmet<'a>(cx: Scope<'a, HelmetProps<'a>>) -> Element {
     None
 }
 
-impl Drop for HelmetProps<'_> {
```
Comment (marc2332) [inline src/lib.rs:96]: Why did you remove this?

File: Cargo.toml [hunk 4]
```diff
@@ -1,18 +1,18 @@
 [package]
 name = "dioxus-helmet"
-version = "0.2.4"
-authors = ["Saicu"]
+version = "0.3.0"
+authors = ["Saicu", "painedpineapple"]
```
Comment (marc2332) [inline Cargo.toml]:
```suggestion
authors = ["Saicu"]
```

File: Cargo.toml [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
 edition = "2021"
 description = "Small Dioxus component which allows you to place elements in the head of your document"
 license = "MIT"
-repository = "https://github.com/saicu/dioxus-helmet"
+repository = "https://github.com/ranger-finance/dioxus-helmet"
```
Comment (marc2332) [inline Cargo.toml]: Wat?

```suggestion
repository = "https://github.com/dioxus-community/dioxus-helmet"
```

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: src/lib.rs [hunk 6]
```diff
@@ -16,32 +16,32 @@
...
-//!                 [r#"
-//!                     body {
-//!                         color: blue;
-//!                     }
-//!                     a {
-//!                         color: red;
-//!                     }
-//!                 "#]
+//!                 r"
+//!                       body {{
```
Comment (marc2332) [inline src/lib.rs:27]: What's with the double bracket?

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 5 above)
Comment (logandeancall) [reply Cargo.toml]: Yeah, sorry, I wasn't originally planning on creating a PR back into this source repo and forgot to change that back.

**Comment by logandeancall** [comment]
@marc2332 My apologies here. I had only intended to use this internally and quickly scraped something that was working for my intended purposes. I should have cleaned it up a bit before opening this PR. I am a bit of a rust noob and am running into an error that I cannot figure out how to resolve when attempting to use the custom HelmetProps with the Drop trait. Maybe this will seem obvious to you?

```
cannot move out of type `HelmetProps`, which implements the `Drop` trait
move occurs because `new_clone.children` has type `std::option::Option<dioxus::prelude::VNode>`, which does not implement the 
```

code:
```rust
#[derive(Props, Clone, PartialEq)]
pub struct HelmetProps {
    children: Element,
}

pub fn Helmet(props: HelmetProps) -> Element {
    if let Some(window) = web_sys::window() {
        if let Some(document) = window.document() {
            if let Some(head) = document.head() {
                if let Some(element_maps) = extract_element_maps(&props.children) {
                    if let Ok(mut init_cache) = INIT_CACHE.try_lock() {
                        element_maps.iter().for_each(|element_map| {
                            let mut hasher = FxHasher::default();
                            element_map.hash(&mut hasher);
                            let hash = hasher.finish();

                            if !init_cache.contains(&hash) {
                                init_cache.push(hash);

                                if let Some(new_element) =
                                    element_map.try_into_element(&document, &hash)
                                {
                                    let _ = head.append_child(&new_element);
                                }
                            }
                        });
                    }
                }
            }
        }
    }

    None
}

impl Drop for HelmetProps {
    fn drop(&mut self) {
        if let Some(window) = web_sys::window() {
            if let Some(document) = window.document() {
                if let Some(element_maps) = extract_element_maps(&self.children) {
                    if let Ok(mut init_cache) = INIT_CACHE.try_lock() {
                        element_maps.iter().for_each(|element_map| {
                            let mut hasher = FxHasher::default();
                            element_map.hash(&mut hasher);
                            let hash = hasher.finish();

                            if let Some(index) = init_cache.iter().position(|&c| c == hash) {
                                init_cache.remove(index);
                            }

                            if let Ok(children) =
                                document.query_selector_all(&format!("[data-helmet-id='{hash}']"))
                            {
                                if let Ok(Some(children_iter)) = js_sys::try_iter(&children) {
                                    children_iter.for_each(|child| {
                                        if let Ok(child) = child {
                                            let el = web_sys::Element::from(child);
                                            el.remove();
                                        };
                                    });
                                }
                            }
                        });
                    }
                }
            }
        }
    }
}
```

File: src/lib.rs (hunk 6 above)
Comment (logandeancall) [reply src/lib.rs:27]: W/out it there is a compiler error:

```
invalid format string: expected `'}'`, found `'2'`
if you intended to print `{`, you can escape it using `{{`
```

**Comment by marc2332** [comment]
Hmmm, I see, it actually doesn't make sense to implement Drop in the Props anyway. I will try to update it according the new dioxus 0.5 apis asap 👍🏻  No worries
